readme.md

```diff
@@ -3,71 +3,704 @@
+
+`"start": "webpack-dev-server --content-base build"`
+
+Go ahead and run `npm start` in your terminal. You should see something like:
+
+![npm Start](npmStart.png)
+
+If you open your browser to `localhost:8080`, you should see your index.html and any html rendered through your index.js file on the browser!
+
+This is cool and all, but not really any different than before other than running a different command in your terminal. We want to be able to make changes to our code and see the updates without having to do any additional refreshing.
+
+We will have to install another package to get this up and running: `npm i webpack-merge --save-dev`
+
+This package allows us to merge objects together. In this case, we will be merging the object that sets our paths, with HMR.
+
+In webpack.config.js, include the following on line 2:
+
+```js
+const merge = require('webpack-merge');
+const webpack = require('webpack');
+const TARGET = process.env.npm_lifecycle_event;
+```
+
+We will then replace everything from `module.exports` and down with the following:
+
+```js
+const common = {
+
+  // Entry accepts a path or an object of entries. We'll be using the
+  // latter form given it's convenient with more complex configurations.
+  entry: {
+    app: PATHS.js
+  },
+  output: {
+    path: PATHS.build,
+    filename: 'bundle.js'
+  }
 };
 
+// Default configuration
+if(TARGET === 'start' || !TARGET) {
+  module.exports = merge(common, {});
+}
+
+if(TARGET === 'build') {
+  module.exports = merge(common, {});
+}
+```
+
+<details>
+<summary>
+What is this doing??
+</summary>
+<br>
+
+```
+Depending on the command we run (start, or build) we will be merging the 'common' object with whatever we include in the empty objects
+```
+<br>
+<br>
+</details>
+
+Now all we are missing is adding in the HMR. Let's edit the first `if` statement to include this:
+
+```js
+if(TARGET === 'start' || !TARGET) {
+  module.exports = merge(common, {
+    devServer: {
+      contentBase: PATHS.build,
+      // Enable history API fallback so HTML5 History API based
+      // routing works. This is a good default that will come
+      // in handy in more complicated setups.
+      historyApiFallback: true,
+      hot: true,
+      inline: true,
+      progress: true,
+      // Display only errors to reduce the amount of output.
+      stats: 'errors-only',
+      // Parse host and port from env so this is easy to customize.
+      // If you use Vagrant or Cloud9, set
+      // host: process.env.HOST || '0.0.0.0';
+      // 0.0.0.0 is available to all network devices unlike default
+      // localhost
+      host: process.env.HOST,
+      port: process.env.PORT
+    },
+    plugins: [
+      new webpack.HotModuleReplacementPlugin()
+    ]
+  });
+}
 ```
 
-## Jenkins
+And remove ` --content-base build` from your `package.json`
+
+**COOL**. Everytime we make a change to the js file and save the browser is automatically updated!
+
+## You Do: Add in CSS watch!
```

Comment:
That's the last part i'm gonna add in still, we just wanted to make sure we gave you/anyone else enough time to look over it today!